jobs/otel-collector-windows/spec

```diff
@@ -16,12 +16,36 @@ properties:
   enabled:
     description: "Enable OTel Collector"
     default: true
-  ingress.grpc.port:
-    description: "Port the collector is listening on to receive OTLP over gRPC"
-    default: 9100
+  config:
```

Comment:
Hi @acrmp,

Generally I'm all in about this change, but I wonder if we should do it this way, that we practically embed the Otel Collector config in the spec or we provide provide a parameter which can define the path to the Otel collector config yaml file.

If we use some manager to manager the fleet of Otel collectors which we have based on OpAMP, they can push configuration to the clients (Otel collectors), it might be more clear, readable, practical for further processing (merge of local and remote config) if the configuration is in a separate file.

Reply:
@chombium Can you expand on how you would see this working? Were you thinking of doing config merging outside of this release?

We're still exploring how we might want to distribute configuration. One option is that we could do something similar to [what we do for the prom scraper and load configs that match a certain file name glob](https://github.com/cloudfoundry/loggregator-agent-release/blob/ba3aa9335b45a077ae68b9bd269323ec44676218/jobs/prom_scraper/spec#L29-L31). A downside is that otel config changes would currently require BOSH to roll all the VMs.

We've looked into OpAMP a little. There still seem to be a few rough edges at the moment. For example see this recent issue we opened:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33799

Reply:
Hi @acrmp, @mkocher sorry for the late reply.... :-/

I was thinking of having the Otel collector config in a separate file for better separation, but it doesn't hurt if it's inside the job spec. At the end if we use an OpAMP server for managing the configuration, the sever would ping the Otel Collector that there is new config available and its up to the collector to decide when it will download and apply the new config. I'm not against your solution at all ;)